openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1318,6 +1341,39 @@ def revert_changes(library_key):
         )
     )
 
+    # For each collection, trigger LIBRARY_COLLECTION_UPDATED signal and set background=True to trigger
+    # collection indexing asynchronously.
+    #
+    # This is to update component counts in all library collections,
+    # because there may be components that have been discarded in the revert.
+    for collection in authoring_api.get_collections(learning_package.id):
```

Comment:
I'm not going to block this PR over this, but I don't think re-indexing every Collection in the Library is an acceptable long term solution. A Library might have many, many Collections. One of the deep problems with Modulestore has been the tendency for apps to have to reprocess the entire course because they don't have granular data on whether the pieces that are relevant to them have changed or not. I don't want that pattern to get started here.

The `reset_drafts_to_published` function can be modified to return a queryset of `Drafts` or `PublishableEntities` or something along those lines, and we can allow this code to figure out which collections were actually updated. (Side note: The `reset_drafts_to_published` is also pretty inefficient because I was only thinking about the manual editing use case and not really thinking about importing potentially thousands of components at once and then hitting "discard changes".)

Reply:
I agree with this, could this be done in a follow-up task? @bradenmacdonald @pomegranited

Reply:
Yes, as long as the bug is fixed now and the fix is backported to Sumac, we can address those refactors and improvements in a follow-up task. Let's do that as soon as sumac.0 is cut. Doing it now will complicate other backports like this one, especially if any change to openedx-learning are required.

But I don't want to postpone it any later than that, or we won't get to it. Ideally you can start on the PRs for it now, even if we don't merge until after Sumac.0 is released.